core/types/transaction_signing.go

```diff
@@ -255,13 +255,13 @@ func decodeSignature(sig []byte) (r, s, v *big.Int) {
 	return r, s, v
 }
 
-func recoverPlain(sighash common.Hash, R, S, Vb *big.Int, homestead bool) (common.Address, error) {
+func recoverPlain(sighash common.Hash, R, S, Vb *big.Int) (common.Address, error) {
 	if Vb.BitLen() > 8 {
 		return common.ZeroAddr, ErrInvalidSig
 	}
 	V := byte(Vb.Uint64() - 27)
-	if !crypto.ValidateSignatureValues(V, R, S, homestead) {
-		return common.ZeroAddr, ErrInvalidSig
+	if !crypto.ValidateSignatureValues(V, R, S) {
+		return common.Address{}, ErrInvalidSig
```

Comment:
no zeroAddr?

Reply:
Technically returning the address 0x0 is wrong because the sig was not signed by that key (nobody has that key)